content/docs/guides/continuous-delivery/troubleshooting-guide.md

```diff
@@ -0,0 +1,122 @@
+a [plugin]({{< relref "/docs/intro/concepts/how-pulumi-works/#resource-providers" >}}) binary from our CDN.
+  * So be sure to cache the plugins path as well.
+  > Note that depending on the number of providers you use in your Pulumi app, the cache size can be huge. 
+  * If in doubt about problems encountered during execution, clear out all caches and restore dependencies from scratch.
+
+## Cloud Provider Credentials
+
+Another common point of failure are the cloud provider credentials. The class of errors related to the credentials can be, but not limited to:
+
+* Incorrect credentials (wrong account, mismatched keys etc.)
+* Keys with a strict access scope that don't have access to creating specific resources
+
+### Tips
+
+* Cloud provider credentials not specified using a recognized mechanism. In almost all cases, specifying the credentials through environment variables will work.
+  * Some cloud providers also support authentication via their respective CLIs. For example, the Azure provider supports authentication via the Azure CLI.
+  * In the above example, if the Azure CLI is not properly authenticated, then Pulumi will not be able to acquire the necessary credentials.
+* Ensure that the Pulumi step has access to the environment variables containing the credentials.
+  * If your CI service has features that allow you to restrict which pipelines or steps in a pipeline can access the secrets, then ensure that
+  your pipeline/steps can access them.
+* Typo in the name of the environment variables. Refer to our [Cloud Providers]({{< relref "/docs/intro/cloud-providers/" >}}) docs and the respective setup pages for each to find the correct
+environment variables to use.
+
+## Other Errors
+
+### Pulumi not found in the `PATH`
+
+This problem is especially prevalent for users using a Docker container-based pipeline. With most Docker container pipelines, the state of the container
+does not persist across steps, so you cannot split-out the installation step of the Pulumi CLI into another step and expect to use the modified environment
+in the following step. There are exceptions to this. Some CI services allow you to create a reusable template step that you can inject as a pre-cursor to other
+steps. This is a way to reduce repeating the installation step every time you would like to run `pulumi` command.  
+
+### Still need help?
+
+If your error is not related to any of the above categories, it is likely that there is a problem with the code in your infrastructure code.
+Our Programming Model docs can be very helpful in understanding the concepts of programming an infrastructure app. If that doesn't help you solve your
+particular issue, then there is always the power of the strong, knowledgable [Pulumi Slack Community](https://slack.pulumi.com). Signup is free and quick. In addition to other
+community members, various members of the Pulumi team themsevles hang out in the Slack channels and are willing to help you.
```

Comment:
themsevles > themselves
maybe "would be happy to help you" instead of "are willing"?

Reply:
Definitely!